syntax = "proto2";
import "state.proto";

message SteeringCmd {
    required int32 steeringSetPoint = 1;
}

message SpeedCmd {
    required int32 speedSetPoint = 1;
}

message WaypointCmd {
    required string longitude = 1;
    required string latitude = 2;
    required int32 sequenceNumber = 3; 
}


enum CmdType {
    STOP = 0;        
}

message BasicCmd {

    required CmdType basicCmd = 1;
}

message ManualControlCmds {
    required int32 steeringSetPoint = 1;
    required int32 speedSetPoint = 2;
    required CarState setCarState = 3;
    required CmdType BasicCmd = 4;
}


message StateCmd {
    required CarState setCarState = 1;
}

message DebugCmd {
    required int32 debugCmd = 1;
}
